birb_registry/src/lib.rs

```diff
@@ -0,0 +1,118 @@
+    }
+}
+
+impl Debug for Registry {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("Registry")
+            .field("data", &self.data)
+            .finish()
+    }
+}
+
+impl MainThreadModule for Registry {
+    fn tick(&mut self, _: &birb::MainThreadApp) {
+        // shitty persistence :3
+        // loads json on zero
+        if self.ticks == 0 {
+            if fs::try_exists("registry.json").unwrap() {
+                match File::open("registry.json") {
+                    Ok(mut File) => {
+                        let mut data = String::new();
+                        match File.read_to_string(&mut data) {
+                            Ok(_) => {
+                                self.data = serde_json::from_str(&data).unwrap()
+                            }
+                            Err(_) => {
+                                fs::remove_file("registry.json").unwrap();
+                                self.ticks = 0;
+                                // early return to escape ++ increment on internal tick counter.
+                                return
+                            }
+                        }
+                    }
+                    Err(E) => {
+                        panic!("error in loading persistence \n {}", E);
+                    }
+                }
+            } else {
+                match File::create("registry.json") {
+                    Ok(mut File) => {
+                        // create from empty Hashmap Array
+                        match File.write(serde_json::to_vec(&self.data).unwrap().as_slice()) {
+                            Err(E) => {
+                                panic!("error saving persistence \n {}", E)
+                            }
+                            _ => {
+                                // discard, dont care about OK() result
+                            }
+                        }
+                    }
+                    Err (E) => {
+                        panic!("error in creating persistence \n {}", E);
+                    }
+                }
+            }
+        } else {
+            // every 5 seconds
+            if self.ticks > 300 {
+                match File::open("registry.json") {
+                    Ok(mut File) => {
+                        match File.write(serde_json::to_vec(&self.data).unwrap().as_slice()) {
+                            Err (E) => {
+                                panic!("error in persisting persistence \n {}", E);
+                            }
+                            _ => {
+                                // discard, dont care about OK() result
+                            }
+                        }
+                    }
+                    Err(E) => {
+                        panic!("error in persisting persistence \n {}", E);
+                    }
+                }
+                self.ticks = 0;
+            }
+        }
+        self.ticks += 1;
```

Comment:
this anchors loading and saving to tick rate, which could vary depending on refresh rate, throttling, etc meaning things won't be saved reliably